### PR TITLE
Zadaca 2

### PR DESCRIPTION
**Pull request by NikolaPehnec**

### CODE REVIEW

File: app/src/main/java/agency/five/tmdb/MovieCategory.kt [hunk 1]
```diff
@@ -20,12 +21,14 @@ import androidx.compose.ui.unit.dp
 fun MovieCategory(
     modifier: Modifier = Modifier,
     categoryModel: MovieCategoryModel,
+    favoriteViewModel: FavoriteMoviesViewModel,
```
Comment (matejsalkovic) [inline app/src/main/java/agency/five/tmdb/MovieCategory.kt:24]: Composable functions that don't represent screens should not know about ViewModel directly, but instead you should pass only state those functions need to render.

File: app/src/main/java/agency/five/tmdb/MainActivity.kt [hunk 2]
```diff
@@ -19,12 +26,24 @@ import androidx.compose.ui.Alignment
...
     @RequiresApi(Build.VERSION_CODES.O)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        startKoin {
+            androidContext(this@MainActivity)
+            modules(repoModule, ViewModelModules, movieApi, mockDB)
+        }
+
+        val homeViewModel: HomeViewModel by viewModel()
```
Comment (matejsalkovic) [inline app/src/main/java/agency/five/tmdb/MainActivity.kt:43]: Move initializations of ViewModels inside screen functions.

File: app/src/main/java/agency/five/tmdb/DI/ViewModelModules.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package agency.five.tmdb.DI
+
+import agency.five.tmdb.viewModel.FavoriteMoviesViewModel
+import agency.five.tmdb.viewModel.HomeViewModel
+import agency.five.tmdb.viewModel.MovieDetailViewModel
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+
+val ViewModelModules = module {
```
Comment (matejsalkovic) [inline app/src/main/java/agency/five/tmdb/DI/ViewModelModules.kt:9]: vals should always start with lowercase

**Review by matejsalkovic — CHANGES_REQUESTED** [review]

File: app/src/main/java/agency/five/tmdb/FavoriteScreen.kt [hunk 4]
```diff
@@ -13,19 +14,24 @@ import androidx.compose.foundation.lazy.LazyVerticalGrid
...
 @OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun FavoriteScreen(
-    favoriteMoviesViewModel: FavoriteMoviesViewModel,
     onMovieCardClick: (String) -> Unit
 ) {
+    val favoriteViewModel: FavoriteMoviesViewModel = getViewModel()
+
+    val favoriteMovies: List<MovieModel> = favoriteViewModel.getFavoriteMovies()
+        .collectAsState(initial = listOf()).value
```
Comment (NikolaPehnec) [inline app/src/main/java/agency/five/tmdb/FavoriteScreen.kt:34]: Not sure if an empty list is a good initial value

File: app/src/main/java/agency/five/tmdb/data/PreviewData.kt [hunk 5]
```diff
@@ -0,0 +1,383 @@
+package agency.five.tmdb.data
+
+import agency.five.tmdb.R
+import android.os.Build
+import androidx.annotation.RequiresApi
+import java.util.*
+
+class PreviewData {
```
Comment (NikolaPehnec) [inline app/src/main/java/agency/five/tmdb/data/PreviewData.kt:8]: I returned the previewData class but don't know how to use it in previews considering the fact that composables don't have Lists of data in constructors

File: app/src/main/java/agency/five/tmdb/FavoriteScreen.kt (hunk 4 above)
Comment (matejsalkovic) [reply app/src/main/java/agency/five/tmdb/FavoriteScreen.kt:34]: It is

File: app/src/main/java/agency/five/tmdb/data/PreviewData.kt (hunk 5 above)
Comment (matejsalkovic) [reply app/src/main/java/agency/five/tmdb/data/PreviewData.kt:8]: You can use the following pattern:
```
@Composable
fun HomeScreen() {
    // viewModel and state
}

@Composable
fun HomeContent(
    // states and click callbacks
) {
     // implement screen here
}

@Preview
@Composable
fun HomePreview() {
    // call HomeContent with preview data
}
```

File: app/src/main/java/agency/five/tmdb/HomeScreen.kt [hunk 6]
```diff
@@ -41,18 +50,19 @@ fun HomeScreen(
...
                     movie.categories.contains(categoryModel.categoryName)
                 }
 
                 MovieCategory(
                     categoryModel = categoryModel,
                     movies = movies,
-                    favoriteViewModel = favoriteViewModel,
-                    onMovieCardClick = onMovieCardClick
-                )
+                    onMovieCardClick = onMovieCardClick,
```
[the remark this answers is not in the document]
Comment (NikolaPehnec) [reply app/src/main/java/agency/five/tmdb/HomeScreen.kt:81]: Not sure how to do that considering the function is called on navController, when calling HomeScreen in NavigationSetup

**Review by matejsalkovic — APPROVED** [review]

**Pull request merged**